### PR TITLE
feat: add groups support to the CLI

### LINKED ISSUE #4414: groups: manage via CLI
Add `coder groups` to CRUD groups and add/remove users.

Not sure if we need this yet, but something to consider for a major release.

### PR DESCRIPTION
**Pull request by sreya**
- Adds a route to get a group by org and name.

fixes #4414

cc @bpmct

### CODE REVIEW

File: enterprise/coderd/coderd.go [hunk 1]
```diff
@@ -77,10 +77,18 @@ func New(ctx context.Context, options *Options) (*API, error) {
 		r.Route("/organizations/{organization}/groups", func(r chi.Router) {
 			r.Use(
 				apiKeyMiddleware,
+				api.templateRBACEnabledMW,
```
Comment (Emyrk) [inline enterprise/coderd/coderd.go:80]: Should `GET` still work? Just making new groups and editing groups disabled?

**Review by Emyrk — COMMENTED** [review]
All seems good 👍 

Only did a quick once over

File: enterprise/cli/groupedit.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+			req.AddUsers, err = convertToUserIDs(addUsers, users)
+			if err != nil {
+				return xerrors.Errorf("parse add-users: %w", err)
+			}
+
+			req.RemoveUsers, err = convertToUserIDs(rmUsers, users)
+			if err != nil {
+				return xerrors.Errorf("parse rm-users: %w", err)
+			}
```
[the remark this answers is not in the document]
Comment (sreya) [reply enterprise/cli/groupedit.go:70]: I think it's fine for now.

**Review by coadler — APPROVED** [review]

File: enterprise/coderd/coderd.go (hunk 1 above)
Comment (sreya) [reply enterprise/coderd/coderd.go:80]: We can do that in the future if it becomes an issue. I'm just maintaining the status quo.

**Pull request merged**